notify/sns/sns.go

```diff
@@ -107,39 +137,35 @@ func (n Notifier) Notify(ctx context.Context, alert ...*types.Alert) (bool, erro
 	if len(n.conf.Attributes) > 0 {
 		attributes := map[string]*sns.MessageAttributeValue{}
 		for k, v := range n.conf.Attributes {
-			attributes[k] = &sns.MessageAttributeValue{DataType: aws.String("String"), StringValue: aws.String(v)}
+			attributes[tmpl(k)] = &sns.MessageAttributeValue{DataType: aws.String("String"), StringValue: aws.String(tmpl(v))}
 		}
 		publishInput.SetMessageAttributes(attributes)
 	}
 
 	if n.conf.Subject != "" {
-		publishInput.SetSubject(n.conf.Subject)
-	}
-
-	// Deduplication key is only added if it's a FIFO SNS Topic.
-	if n.conf.IsFIFOTopic {
-		key, err := notify.ExtractGroupKey(ctx)
-		if err != nil {
-			return false, err
-		}
-		publishInput.SetMessageDeduplicationId(key.Hash())
+		publishInput.SetSubject(tmpl(n.conf.Subject))
 	}
 
 	publishOutput, err := client.Publish(publishInput)
 	if err != nil {
-		// AWS Response is bad, probably a config issue.
-		return false, err
+		return n.retrier.Check(err.(awserr.RequestFailure).StatusCode(), strings.NewReader(err.(awserr.RequestFailure).Message()))
```

Comment:
If the error is not a `awserr.RequestFailure` then it will panic. To have the code more robust, you should check for the data type first.

Reply:
In the event this isn't an awserr, should we be using a retrier? or just `return true, err`

Reply:
just return. In the next few seconds, this would be retried by other instances in the HA pair.